azurerm/internal/services/datafactory/resource_arm_data_factory_linked_service_blob_storage.go

```diff
@@ -0,0 +1,255 @@
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory Linked Service BlobStorage Anonymous %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_linked_service_blob_storage", *existing.ID)
+		}
+	}
+
+	connectionString := d.Get("connection_string").(string)
+	secureString := datafactory.SecureString{
+		Value: &connectionString,
+		Type:  datafactory.TypeSecureString,
+	}
+
+	blobProperties := &datafactory.AzureBlobStorageLinkedServiceTypeProperties{
+		ConnectionString: &secureString,
+	}
+
+	description := d.Get("description").(string)
+
+	blobStorageLinkedService := &datafactory.AzureBlobStorageLinkedService{
+		Description: &description,
+		AzureBlobStorageLinkedServiceTypeProperties: blobProperties,
+		Type: datafactory.TypeAzureBlobStorage,
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		blobStorageLinkedService.Parameters = expandDataFactoryParameters(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("integration_runtime_name"); ok {
+		blobStorageLinkedService.ConnectVia = expandDataFactoryLinkedServiceIntegrationRuntime(v.(string))
+	}
+
+	if v, ok := d.GetOk("additional_properties"); ok {
+		blobStorageLinkedService.AdditionalProperties = v.(map[string]interface{})
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		blobStorageLinkedService.Annotations = &annotations
+	}
+
+	linkedService := datafactory.LinkedServiceResource{
+		Properties: blobStorageLinkedService,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, dataFactoryName, name, linkedService, ""); err != nil {
+		return fmt.Errorf("Error creating/updating Data Factory Linked Service BlobStorage %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Data Factory Linked Service BlobStorage %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Data Factory Linked Service BlobStorage %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryLinkedServiceBlobStorageRead(d, meta)
+}
+
+func resourceArmDataFactoryLinkedServiceBlobStorageRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["linkedservices"]
```

Comment:
Could we write a parse ID function for this like in #5356?

Reply:
Is this parsing an azure resource ID string? This seems like there would already be a utility function outside the scope of data factory resources. Maybe I’m misunderstanding what the purpose of said function is... I haven’t reviewed the issue referenced